### DISCUSSION TITLE
How can I get current chain Id which I connected.

### DISCUSSION THREAD

**Discussion by almatutor** [wevm/wagmi#3759] [Q&A]
I hope to know how can I get current chain id which I connected via metamask or coinbase wallet.
I tried to use useChainId function, but it returns Sepolia chain even if I switched network to Polygon in metamask. 

```
const chainId = useChainId()
```

Here is wagmi setting that I set and I doubt the problem is because of that. 

```
import { http, createConfig } from 'wagmi'
import { mainnet, sepolia } from 'wagmi/chains'
import { coinbaseWallet, walletConnect, metaMask } from 'wagmi/connectors'

export const config = createConfig({
  chains: [mainnet, sepolia],
  connectors: [
    metaMask(),
    coinbaseWallet({ appName: 'Create Wagmi' }),
    walletConnect({ projectId: import.meta.env.VITE_WC_PROJECT_ID }),
  ],
  transports: {
    [mainnet.id]: http(),
    [sepolia.id]: http(),
  },
})

declare module 'wagmi' {
  interface Register {
    config: typeof config
  }
}
```
So, when I switch network to Polygon on metamask, how to detect it so that it can get Polygon instead of Sepolia as current chain ID?

**Comment by tmm** [comment]
https://wagmi.sh/react/api/hooks/useAccount#chainid

**Comment by Maurechi** [comment]
cause you haven't configured polygon on your wagmi config.
